enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -0,0 +1,774 @@
+---
+title: microshift-updateability-ostree
+authors:
+  - "@pmtk"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@pacevedom, MicroShift team"
+  - "@ggiguash, MicroShift team"
+  - "@copejon, MicroShift team"
+  - "@dusk125, etcd team"
+  - "@oglok, author of previous enhancement"
+  - "RHEL expert"
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - "@dhellmann"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2023-04-14
+last-updated: 2023-04-14
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/USHIFT-518
+see-also:
+  - "/enhancements/microshift/microshift-greenboot.md"
+  - "/enhancements/microshift/etcd-supportability.md"
+replaces:
+  - https://github.com/openshift/enhancements/pull/1312
+superseded-by:
+  - None
+---
+
+# MicroShift updateability in ostree based systems
+
+## Summary
+
+This enhancement focuses on high level overview of updating
+MicroShift running on ostree based systems such as RHEL 4 Edge.
+Enhancement covers backup and restore of MicroShift data,
+version migration (upgrade and downgrade) of MicroShift and 
+its consequences (migration of data between schema versions),
+and interactions with GreenBoot and operating system.
+
+## Motivation
+
+MicroShift team is working towards a general availability (GA) release.
+As GA product, it is expected that it can be updated to
+provide security patches, functional updates, and bug fixes
+without needing to redeploy.
+
+MicroShift is intended to be a part of Red Hat Device Edge
+which is based on RHEL For Edge which features ostree and as such
+provides rollbacks to go back to previous ostree deployment.
+Even though, OpenShift does not support downgrade or rollback,
```

Comment:
There has been some discussion about having OCP support downgrading to an earlier z-stream release within the same y-stream. Is there any reason we couldn't do the same for MicroShift? Would it be useful to support that?

@LalatenduMohanty do you have any references related to that z-stream rollback work that you can share?

Reply:
I think it would, I hope we don't experience 4.12.7 fiasco again, but I think we cannot rule out that rebase introduces some bug

Reply:
Can we guarantee that storage versions stay the same within whole Y stream?
If that's the case, then this shouldn't be a big problem, we just loosen the logic to ignore Z stream so we can move freely across whole Y stream

Reply:
We do not make API-breaking changes within a Y stream, so the storage version should stay consistent.

Reply:
[OTA-455](https://issues.redhat.com/browse/OTA-455) is tracking rollbacks within a z-stream for OCP.  It's mostly waiting on consensus for how to respond when something lands that breaks those rollbacks (e.g. openshift/release#22859) or our test coverage of those rollbacks (e.g. openshift/release#25311).